### DISCUSSION TITLE
Is Volar the new version of Vetur?

### DISCUSSION THREAD

**Discussion by JenuelDev** [vuejs/language-tools#583] [Q&A] [answered]
is Volar the new version for Vue development? do we still need to use vetur or change to volar?

**Comment by ghost** [comment, marked as answer]
Evan You during the Vue Contributor Days a few days ago said that the quick start guide for Vue will now recommend Volar instead of Vetur. He said it here: https://youtu.be/gpTbH469Qog?t=1926

EDIT: ... and more recently here https://twitter.com/youyuxi/status/1458004347854356484

Relevant Vite PR created by vite & vue team member patak-js: https://github.com/vitejs/vite/pull/4842

I interpret it like "If you're staying with Vue 2 OR if you don't use TypeScript that much - Vetur is still fine, but if you're going to be using Vue 3 or you use typescript extensively pick Volar, as this is the main tool that'll be used by the ecosystem going forward"

The reason for my TypeScript interpretation is that Vetur doesn't have vue-tsc etc., so no type-checking in the template and so on

**Reply by JenuelDev** [reply]
I was curious because I am actually using vetur in my vue 3 CLI project using TypeScript, and typescript works really fine... I really don't know why volar though... But is Vetur going to upgrade to support what volar can do that vetur cant?

**Reply by ghost** [reply]
> and typescript works really fine

That depends how much you expect from TypeScript in your project. If you want basic type information Vetur is still good, but if you want TypeScript LSP code actions, type checking in the template (more complex than basic types "string", "boolean" - Volar detects Vue 3's `as PropType`) etc. that works better in Volar.

Vetur also doesn't support `<script setup>` if I remember correctly. 

> But is Vetur going to upgrade to support what volar can do that vetur cant

We can't know that :P If you use VSCode this is just a matter of changing extensions so the cost of switching is low. 

> I really don't know why volar though

If you don't know what to pick, just pick Volar. It is/will be the default recommendated extension. Evan said he uses Volar himself most of the time these days (during Nuxt conference a few weeks ago) for Vue 3 development. I'd trust the Vue team's recommendation on this, even if it doesn't make sense to you for now. You may not be using `<script setup>` yet or not using code actions a lot (eg. rename) where Volar really shines.

**Reply by JenuelDev** [reply]
I just really don't like to have many extensions in my vscode, it takes resources of my computer... But thanks for answering, I think I will start using volar now then... one last question volar supports `vue 3` and `vue 2` right? and it does support `vue-CLI`, `nuxt 3` and` Vite`?

**Reply by ghost** [reply]
> one last question volar supports vue 3 and vue 2 right

https://github.com/johnsoncodehk/volar/blob/master/extensions/vscode-vue-language-features/README.md?plain=1#L28-L63

>and it does support vue-CLI, nuxt 3 and Vite

It does. I don't think Volar's functionality is affected by which build tools/frameworks you use. Volar just cares about .vue files it's viewing

Nuxt even recommends Volar in its readme: https://github.com/nuxt/framework/blob/950d08559ab4a2605bff1b159399ac9573a836d2/docs/content/1.getting-started/1.introduction.md?plain=1#L20

There's also the upcoming https://github.com/vuejs/create-vue which will replace Vue CLI, but that also shouldn't affect Volar

**Reply by franciscolourenco** [reply]
Any features missing from Volar that Vetur has? Anyone has a feature comparison table?

**Reply by Super-Chama** [reply]
volar seems very slow to start (~365ms) I hope they fix this. vetur is much fast

**Reply by johnsoncodehk** [reply]
@Super-Chama This time is inaccurate 😅, it does not take into account the time it takes for Vetur to unpack dependencies.

**Reply by spf-oss** [reply]
For vue2 projects, can we completely replace vetur with volar?

**Reply by johnsoncodehk** [reply]
@spf-oss for vue < 2.7 need some config (see https://github.com/johnsoncodehk/volar/tree/master/extensions/vscode-vue-language-features#usage), for vue 2.7 yes.
